### ISSUE TITLE
Cannot start ASWebAuthenticationSession without providing presentation context

### ISSUE THREAD

**Issue by psredzinski** [AzureAD/microsoft-authentication-library-for-objc#588]
I am encountering this issue after updating iPhone to iOS 13 and macOS to Catalina. I know that these are beta versions, but I'd love to have MSAL working properly

**Comment by oldalton** [comment]
@psredzinski, we're currently analysing all changes introduced in iOS 13, including this one. @antrix1989, assigning you to this one.

**Comment by psredzinski** [comment]
Do you have any idea when could I expect this to be fixed and released? If it is a week or months?

**Comment by antrix1989** [comment]
hi @psredzinski, this should be available in 1 week.

**Comment by mikepugh** [comment]
@antrix1989 any updates on this? Thanks!

**Comment by mosess** [comment]
@antrix1989 is there a plan to resolve this issue soon?

**Comment by antrix1989** [comment]
Please use this [branch](https://github.com/AzureAD/microsoft-authentication-library-for-objc/tree/sedemche/dev_xcode11) to test the updated SDK with iOS 13 changes.

**Comment by mikepugh** [comment]
@antrix1989 I tried updating my project's pod to:

`pod 'MSAL', :git => 'https://github.com/AzureAD/microsoft-authentication-library-for-objc.git', :branch => 'dev_xcode11'`

However now I'm getting a build error:

`'IdentityCore_Internal.h' file not found in MSAL_Internal.h`

I also noticed when I installed the pod from that branch, it listed MSAL as version `0.4.3`

**Comment by oldalton** [comment]
@mikepugh, you need to add  :submodules => true to your pod file, like:

`pod 'MSAL', :git => 'https://github.com/AzureAD/microsoft-authentication-library-for-objc.git', :branch => 'dev_xcode11', :submodules => true`

**Comment by mikepugh** [comment]
@oldalton thanks for that! I'm getting another error now after re-installing the pod with submodules:

`Implicit declaration of function 'MSID_LOG_INFO' is invalid in C99` on line 109 of `MSIDSFAuthenticationSession.m`

I'm going to try starting with a new blank project, just in case I've messed something up along the way.

**Comment by mikepugh** [comment]
I tried two new empty Single View App iOS projects - one with SwiftUI and one without. Both result in the same compile error after I install the `dev_xcode11` version of this pod.  

I'm running Xcode 11 beta 5 on Catalina 19A526h

**Comment by oldalton** [comment]
`Implicit declaration of function 'MSID_LOG_INFO' is invalid in C99 on line 109 of MSIDSFAuthenticationSession.m`

@antrix1989, looks like it might be caused by our logging changes. I think we should merge latest dev into dev_xcode11 branch. Can you take a look?

**Comment by antrix1989** [comment]
MSIDSFAuthenticationSession isn't used in the project, seems it was added accidentally. Removed it from the repo.  

@mikepugh please try again, it should work now.

**Comment by mikepugh** [comment]
@antrix1989 thanks, but I'm getting another build error now.

File: `MSALPublicClientApplication.m` line 34 `#import "MSIDMacKeychainTokenCache.h"`

Error: `'MSIDMacKeychainTokenCache.h' file not found`

**Comment by antrix1989** [comment]
@mikepugh did you add submodules flag? Like here:

`pod 'MSAL', :git => 'https://github.com/AzureAD/microsoft-authentication-library-for-objc.git', :branch => 'dev_xcode11', :submodules => true`

**Comment by antrix1989** [comment]
@mikepugh updated, please try again.

**Comment by mikepugh** [comment]
@antrix1989 it's compiling now, thanks. I'll see if I can get this working in my SwiftUI app later this weekend or next week.

**Comment by Parameswaran-Irulandi** [comment]
@antrix1989 When do you merge with develop?.Since we are planing to migrate iOS13 version

**Comment by antrix1989** [comment]
Hi @Parameswaran-Irulandi, I can not repro this. Are you testing on Mac or iOS? Could you please send your MSAL logs to my email? sedemche@microsoft.com

**Comment by Parameswaran-Irulandi** [comment]
Thanks @antrix1989 .I already send MSAL logs to your email.

**Comment by Parameswaran-Irulandi** [comment]
@antrix1989 
 
Earlier, I didn’t add keychain sharing Signing &Capabilities. After I add the keychain Groups with ‘com.microsoft.adalcache’
 
It doesn’t crash. Is it expected?

**Comment by tspenc-legrand** [comment]
> @Parameswaran-Irulandi it should be merged into the dev branch in 1-2 weeks.

Do you have a release for this already or an eta for release?  We are anxiously looking forward to this as we are having problems logging into our app using iOS 13, and consequently cannot test other iOS 13 changes.

**Comment by Parameswaran-Irulandi** [comment]
@tspenc-legrand I am also waiting for @antrix1989  for merging this branch to develop.

I try to use this branch with our code base.It works .....

pod 'MSAL', :git => 'https://github.com/AzureAD/microsoft-authentication-library-for-objc.git', :branch => 'dev_xcode11', :submodules => true

**Comment by mikepugh** [comment]
Just in case this helps anyone else - I was able to get everything wired up and working in iOS 13 using the latest version (0.7.1) of MSAL and SwiftUI once I found out how to pass in the view controller:

```
let keyWindow = UIApplication.shared.connectedScenes.filter({$0.activationState == .foregroundActive}).map({$0 as? UIWindowScene}).compactMap({$0}).first?.windows.filter({$0.isKeyWindow}).first

let rvc = keyWindow?.rootViewController

let parameters = MSALInteractiveTokenParameters(scopes: [GraphScopes.UserRead.rawValue, GraphScopes.CalendarsRead.rawValue], webviewParameters: MSALWebviewParameters(parentViewController: rvc!))
```

**Comment by oldalton** [comment]
Should be fixed in the latest MSAL release (0.7.0+). Closing the issue.
